### ISSUE TITLE
[1.18.1 Forge] Crashing Server on Startup

### ISSUE THREAD

**Issue by red2blue** [lucko/spark#147]
39.0.7 modloader

also reported to Architectury https://github.com/issues?q=is%3Aissue+author%3Ared2blue+archived%3Afalse+is%3Aclosed
```java
[06:27:26] [main/ERROR]: Exception caught during firing event: net/minecraftforge/server/permission/DefaultPermissionLevel
        Index: 7
        Listeners:
                0: HIGH
                1: ASM: class dev.architectury.event.forge.EventHandlerImplCommon event(Lnet/minecraftforge/event/RegisterCommandsEvent;)V
                2: NORMAL
                3: ASM: class me.ichun.mods.deathcounter.common.core.DeathHandler onRegisterCommands(Lnet/minecraftforge/event/RegisterCommandsEvent;)V
                4: ASM: class com.sunekaer.mods.toolkit.commands.TKCommand registerCommands(Lnet/minecraftforge/event/RegisterCommandsEvent;)V
                5: ASM: com.refinedmods.refinedstorage.setup.ServerSetup@42926b1d onRegisterCommands(Lnet/minecraftforge/event/RegisterCommandsEvent;)V
                6: net.minecraftforge.eventbus.EventBus$$Lambda$3697/0x00000008015a35a0@5f1d75a1
                7: ASM: me.lucko.spark.forge.ForgeSparkMod@73aab393 registerCommands(Lnet/minecraftforge/event/RegisterCommandsEvent;)V
                8: net.minecraftforge.eventbus.EventBus$$Lambda$3697/0x00000008015a35a0@3d8f071a
                9: ASM: net.xalcon.torchmaster.Torchmaster@45fedf15 onRegisterCommands(Lnet/minecraftforge/event/RegisterCommandsEvent;)V
                10: ASM: class ovh.corail.woodcutter.command.CommandWoodcutter onRegisterCommands(Lnet/minecraftforge/event/RegisterCommandsEvent;)V
                11: ASM: class com.sekwah.reskin.ReSkin onServerStarting(Lnet/minecraftforge/event/RegisterCommandsEvent;)V
                12: ASM: atomicstryker.infernalmobs.common.InfernalMobsCore@7c12a8b9 registerCommands(Lnet/minecraftforge/event/RegisterCommandsEvent;)V
                13: ASM: class ovh.corail.tombstone.event.EventHandler onRegisterCommand(Lnet/minecraftforge/event/RegisterCommandsEvent;)V
                14: ASM: class net.mehvahdjukaar.supplementaries.setup.ModSetup registerCommands(Lnet/minecraftforge/event/RegisterCommandsEvent;)V
                15: net.minecraftforge.eventbus.EventBus$$Lambda$3697/0x00000008015a35a0@4ebdfdcd
                16: net.minecraftforge.eventbus.EventBus$$Lambda$3697/0x00000008015a35a0@65cab196
                17: ASM: class com.ldtteam.structurize.event.EventSubscriber onRegisterCommands(Lnet/minecraftforge/event/RegisterCommandsEvent;)V
                18: net.minecraftforge.eventbus.EventBus$$Lambda$3697/0x00000008015a35a0@51ce60f3
                19: ASM: class de.srendi.advancedperipherals.common.commands.APCommands register(Lnet/minecraftforge/event/RegisterCommandsEvent;)V
                20: net.minecraftforge.eventbus.EventBus$$Lambda$3697/0x00000008015a35a0@14bc0a07
                21: net.minecraftforge.eventbus.EventBus$$Lambda$3697/0x00000008015a35a0@35cedec8
                22: net.minecraftforge.eventbus.EventBus$$Lambda$3697/0x00000008015a35a0@5388fe6
                23: net.minecraftforge.eventbus.EventBus$$Lambda$3697/0x00000008015a35a0@27d3be4a
                24: ASM: class gollorum.signpost.minecraft.registry.CommandRegistry onRegisterCommands(Lnet/minecraftforge/event/RegisterCommandsEvent;)V
                25: ASM: class dan200.computercraft.shared.CommonHooks onRegisterCommand(Lnet/minecraftforge/event/RegisterCommandsEvent;)V
                26: ASM: com.lothrazar.cyclic.registry.CommandRegistry@29607f61 onRegisterCommandsEvent(Lnet/minecraftforge/event/RegisterCommandsEvent;)V
                27: net.minecraftforge.eventbus.EventBus$$Lambda$3697/0x00000008015a35a0@534b6485
                28: net.minecraftforge.eventbus.EventBus$$Lambda$3697/0x00000008015a35a0@10b0c8ba
                29: ASM: stepsword.mahoutsukai.handlers.ServerHandler@16c342ae serverLoad(Lnet/minecraftforge/event/RegisterCommandsEvent;)V
                30: net.minecraftforge.eventbus.EventBus$$Lambda$3697/0x00000008015a35a0@513622f5
                31: ASM: net.minecraftforge.common.ForgeInternalHandler@523a66f onCommandsRegister(Lnet/minecraftforge/event/RegisterCommandsEvent;)V
                32: net.minecraftforge.eventbus.EventBus$$Lambda$3697/0x00000008015a35a0@6bade777
                33: ASM: class com.minecolonies.coremod.event.EventHandler onCommandsRegister(Lnet/minecraftforge/event/RegisterCommandsEvent;)V
                34: ASM: class com.simibubi.create.events.CommonEvents registerCommands(Lnet/minecraftforge/event/RegisterCommandsEvent;)V
                35: net.minecraftforge.eventbus.EventBus$$Lambda$3697/0x00000008015a35a0@63cad9cb
                36: net.minecraftforge.eventbus.EventBus$$Lambda$3697/0x00000008015a35a0@75292810
                37: ASM: com.blakebr0.mysticalcustomization.command.ModCommands@aaab19e onRegisterCommands(Lnet/minecraftforge/event/RegisterCommandsEvent;)V
                38: ASM: com.blakebr0.cucumber.command.ModCommands@38e6150e onRegisterCommands(Lnet/minecraftforge/event/RegisterCommandsEvent;)V
java.lang.NoClassDefFoundError: net/minecraftforge/server/permission/DefaultPermissionLevel
        at TRANSFORMER/spark@1.6.3/me.lucko.spark.forge.plugin.ForgeServerSparkPlugin.register(ForgeServerSparkPlugin.java:70)
        at TRANSFORMER/spark@1.6.3/me.lucko.spark.forge.ForgeSparkMod.registerCommands(ForgeSparkMod.java:66)
        at net.minecraftforge.eventbus.ASMEventHandler_74_ForgeSparkMod_registerCommands_RegisterCommandsEvent.invoke(.dynamic)
        at MC-BOOTSTRAP/eventbus@5.0.3/net.minecraftforge.eventbus.ASMEventHandler.invoke(ASMEventHandler.java:85)
        at MC-BOOTSTRAP/eventbus@5.0.3/net.minecraftforge.eventbus.EventBus.post(EventBus.java:302)
        at MC-BOOTSTRAP/eventbus@5.0.3/net.minecraftforge.eventbus.EventBus.post(EventBus.java:283)
        at TRANSFORMER/forge@39.0.7/net.minecraftforge.event.ForgeEventFactory.onCommandRegister(ForgeEventFactory.java:730)
        at TRANSFORMER/minecraft@1.18.1/net.minecraft.commands.Commands.<init>(Commands.java:202)
        at TRANSFORMER/minecraft@1.18.1/net.minecraft.server.ServerResources.<init>(ServerResources.java:35)
        at TRANSFORMER/minecraft@1.18.1/net.minecraft.server.ServerResources.m_180005_(ServerResources.java:84)
        at TRANSFORMER/minecraft@1.18.1/net.minecraft.server.Main.main(Main.java:147)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at MC-BOOTSTRAP/fmlloader@1.18.1-39.0.7/net.minecraftforge.fml.loading.targets.CommonServerLaunchHandler.lambda$launchService$0(CommonServerLaunchHandler.java:46)
        at MC-BOOTSTRAP/cpw.mods.modlauncher@9.0.7/cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:37)
        at MC-BOOTSTRAP/cpw.mods.modlauncher@9.0.7/cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:53)
        at MC-BOOTSTRAP/cpw.mods.modlauncher@9.0.7/cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71)
        at MC-BOOTSTRAP/cpw.mods.modlauncher@9.0.7/cpw.mods.modlauncher.Launcher.run(Launcher.java:106)
        at MC-BOOTSTRAP/cpw.mods.modlauncher@9.0.7/cpw.mods.modlauncher.Launcher.main(Launcher.java:77)
        at MC-BOOTSTRAP/cpw.mods.modlauncher@9.0.7/cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26)
        at MC-BOOTSTRAP/cpw.mods.modlauncher@9.0.7/cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23)
        at cpw.mods.bootstraplauncher@0.1.17/cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:90)
Caused by: java.lang.ClassNotFoundException: net.minecraftforge.server.permission.DefaultPermissionLevel
        at cpw.mods.securejarhandler@0.9.54/cpw.mods.cl.ModuleClassLoader.loadClass(ModuleClassLoader.java:135)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
        ... 24 more

Exception in thread "main" java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
        at MC-BOOTSTRAP/cpw.mods.modlauncher@9.0.7/cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:39)
        at MC-BOOTSTRAP/cpw.mods.modlauncher@9.0.7/cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:53)
        at MC-BOOTSTRAP/cpw.mods.modlauncher@9.0.7/cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71)
        at MC-BOOTSTRAP/cpw.mods.modlauncher@9.0.7/cpw.mods.modlauncher.Launcher.run(Launcher.java:106)
        at MC-BOOTSTRAP/cpw.mods.modlauncher@9.0.7/cpw.mods.modlauncher.Launcher.main(Launcher.java:77)
        at MC-BOOTSTRAP/cpw.mods.modlauncher@9.0.7/cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26)
        at MC-BOOTSTRAP/cpw.mods.modlauncher@9.0.7/cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23)
        at cpw.mods.bootstraplauncher@0.1.17/cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:90)
Caused by: java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at MC-BOOTSTRAP/fmlloader@1.18.1-39.0.7/net.minecraftforge.fml.loading.targets.CommonServerLaunchHandler.lambda$launchService$0(CommonServerLaunchHandler.java:46)
        at MC-BOOTSTRAP/cpw.mods.modlauncher@9.0.7/cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:37)
        ... 7 more
Caused by: java.lang.NoClassDefFoundError: net/minecraftforge/server/permission/DefaultPermissionLevel
        at TRANSFORMER/spark@1.6.3/me.lucko.spark.forge.plugin.ForgeServerSparkPlugin.register(ForgeServerSparkPlugin.java:70)
        at TRANSFORMER/spark@1.6.3/me.lucko.spark.forge.ForgeSparkMod.registerCommands(ForgeSparkMod.java:66)
        at net.minecraftforge.eventbus.ASMEventHandler_74_ForgeSparkMod_registerCommands_RegisterCommandsEvent.invoke(.dynamic)
        at MC-BOOTSTRAP/eventbus@5.0.3/net.minecraftforge.eventbus.ASMEventHandler.invoke(ASMEventHandler.java:85)
        at MC-BOOTSTRAP/eventbus@5.0.3/net.minecraftforge.eventbus.EventBus.post(EventBus.java:302)
        at MC-BOOTSTRAP/eventbus@5.0.3/net.minecraftforge.eventbus.EventBus.post(EventBus.java:283)
        at TRANSFORMER/forge@39.0.7/net.minecraftforge.event.ForgeEventFactory.onCommandRegister(ForgeEventFactory.java:730)
        at TRANSFORMER/minecraft@1.18.1/net.minecraft.commands.Commands.<init>(Commands.java:202)
        at TRANSFORMER/minecraft@1.18.1/net.minecraft.server.ServerResources.<init>(ServerResources.java:35)
        at TRANSFORMER/minecraft@1.18.1/net.minecraft.server.ServerResources.m_180005_(ServerResources.java:84)
        at TRANSFORMER/minecraft@1.18.1/net.minecraft.server.Main.main(Main.java:147)
        ... 13 more
Caused by: java.lang.ClassNotFoundException: net.minecraftforge.server.permission.DefaultPermissionLevel
        at cpw.mods.securejarhandler@0.9.54/cpw.mods.cl.ModuleClassLoader.loadClass(ModuleClassLoader.java:135)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
        ... 24 more
Exception in thread "Thread-4" java.lang.NullPointerException: Cannot invoke "java.lang.Thread.interrupt()" because "this.thread" is null
        at TRANSFORMER/simplerpc@1.18-version/me.hypherionmc.simplerpclib.util.FileWatcherService.stop(FileWatcherService.java:69)
        at java.base/java.lang.Thread.run(Thread.java:833)
        ```

**Comment by josiah-roberts** [comment]
I have the same issue.

**Comment by lucko** [comment]
Thanks for reporting, this is fixed by 0def5ca6b5c78f7afa7e728129612f5878dd8209

I've pushed a new release to CurseForge with the fix.

**Comment by josiah-roberts** [comment]
Thanks so much - working now!
